File: Java/12. Reverse of a number.java
import java.util.Scanner;

class Main {
    public static void main(String [] args) {
        Scanner userInput = new Scanner(System.in);

        System.out.println("Enter the Number to Reverse:");
        int enteredNumber = userInput.nextInt();
        int originalNumber = enteredNumber;
        int reverse = 0;
        int remaining;

        while(enteredNumber != 0) {
            remaining = enteredNumber % 10;
            reverse = reverse * 10 + remaining;
            enteredNumber /= 10;
        }
        System.out.println("The reverse of the number is :" + reverse );

        // num => str convertion
        String numberToString = String.valueOf(originalNumber);
        System.out.println(new StringBuilder(numberToString).reverse().toString());
        userInput.close();
    }
}

File: Python/12. Reverse of a number.py
digit = int(input("Enter the Number to Reverse:"))
originalDigit = digit #this line is used for providing the same digit value to the num to str convertion(because the digit is modified each time by while loop)
reverse = 0
remaining = 0

while ( digit != 0 ):
    remaining = digit % 10
    reverse = reverse * 10 + remaining
    digit //= 10

print("The Reverse of the Number is", reverse)

# Simple int -> str convert method
print(str(originalDigit)[::-1])